code-injection-analysis/InitVigilance_jamieDSOTesting/CWE_079_094_116.js

```
@@ -0,0 +1,161 @@
+
+// should trigger codeql/javascript/ql/src/Security/CWE-094/ImproperCodeSanitization.ql
+
+function createObjectWrite() {
+    const assignment = `obj[${JSON.stringify(key)}]=42`;
+    return `(function(){${assignment}})` // NOT OK
+}
+
+// should trigger codeql/javascript/ql/src/Security/CWE-094/UnsafeCodeConstruction.ql
+
+export function unsafeDeserialize(value) {
+  return eval(`(${value})`);
```

Biome (2.1.2)

[error] eval() exposes to security risks and performance issues.

See the MDN web docs for more details.
Refactor the code so that it doesn't need to call eval().

(lint/security/noGlobalEval)